Document/0x05b-Basic-Security_Testing.md

```diff
@@ -362,60 +369,68 @@ Comprehensive documentation, including an installation guide, tutorials, and usa
+For this section we assume that you've properly enabled Developer Mode and USB debugging as explained in "Testing on a Real Device". Once you've connected your Android device via USB, you can access the remote device's shell by running:
 
 ```shell
-$ adb devices
-List of devices attached
-BAZ5ORFARKOZYDFA    device
+$ adb shell
 ```
 
-Access to the remote device's shell
+> press Control + D or type `exit` to quit
+
+If your device is rooted or you're using the emulator, you can get root access by running `su` once in the remote shell:
 
 ```shell
 $ adb shell
+generic_x86:/ $ su
+generic_x86:/ # id
+uid=0(root) gid=0(root) groups=0(root),... context=u:r:su:s0
+
+$ adb shell
+bullhead:/ $ su
+bullhead:/ # id
+uid=0(root) gid=0(root) groups=0(root) context=u:r:su:s0
 ```
 
-> When you are ready to exit the remote shell, press Control + D or type `exit`.
+> Only if you're working with an emulator you may alternatively restart adb with root permissions with the command `adb root` so next time you enter `adb shell` you'll have root access already and it also allows to transfer data from/to file system locations where only the root user has access to (via `adb push/pull`). See more about data transfer in section "Host-Device Data Transfer" below.
 
-This will simply work if you have only one device connected. If you have more than one, you'll have to specify the device to which you want to access the shell.
+###### Connect to Multiple Devices
 
-The `adb shell` command can also issue commands to the device without entering the adb remote shell, for example:
+If you have more than one device, remember to include the `-s` flag followed by the device serial ID on all your `adb` commands (e.g. `adb -s emulator-5554 shell` or `adb -s 00b604081540b7c6 shell`). You can get a list of all connected devices and their serial IDs by using the following command:
 
 ```shell
-$ adb shell id
+$ adb devices
+List of devices attached
+00c907098530a82c    device
+emulator-5554    device
 ```
 
-###### Connecting to an Android Virtual Device
+###### Connect to a Device over Wi-Fi
 
-You can either start an Android Virtual Device (AVD) by using the AVD Manager in Android Studio or start the AVD manager from the command line with the `android` command, which is found  in the tools directory of the Android SDK:
+You can also access your Android device without using the USB cable. For this you'll have to connect both your host computer and your Android device to the same Wi-Fi network and follow the next steps:
 
-```shell
-$ ./android avd
-```
+- Connect the device to the host computer with a USB cable and set the target device to listen for a TCP/IP connection on port 5555: `adb tcpip 5555`.
+- Disconnect the USB cable from the target device and run `adb connect <device_ip_address>`. Check that the device is now available by running `adb devices`.
+- Open the shell with `adb shell`.
```

Comment:
Maybe we should mention that everybody who knows the IP address will also have access to the device and connecting via USB is the preferred way ;-)

Reply:
added after the list: "However, notice that by doing this you leave your device open to anyone being in the same network and knowing the IP address of your device. You may rather prefer using the USB connection."